src/Infrastructure/LeanCode.Npgsql.ActiveDirectory/NpgsqlDataSourceBuilderExtensions.cs

```diff
@@ -0,0 +1,27 @@
+using Azure.Core;
+using Npgsql;
+
+namespace LeanCode.Npgsql.ActiveDirectory;
+
+public static class NpgsqlDataSourceBuilderExtensions
+{
+    public static NpgsqlDataSourceBuilder UseAzureActiveDirectoryAuthentication(
+        this NpgsqlDataSourceBuilder builder,
+        TokenCredential credential
+    )
+    {
+        return builder.UsePeriodicPasswordProvider(
+            async (_, ct) =>
+            {
+                var token = await credential.GetTokenAsync(
+                    new(new[] { "https://ossrdbms-aad.database.windows.net" }, null),
```

Comment:
Locally some analyzer complained to me that this `new[] { "https://ossrdbms-aad.database.windows.net" }` should be a `private static readonly string[]` field for performance reasons.

Reply:
Yeah I saw it too, but TBH I don't have any guarantees regarding how this array is consumed. I _expect_ it not to be modified, but I'm not sure.

Reply:
I think we can safely assume that :)